src/VecSim/spaces/IP/IP_SVE_UINT8.h

```diff
@@ -0,0 +1,101 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+#pragma once
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+static void InnerProductStep(const uint8_t *&pVect1, const uint8_t *&pVect2, size_t &offset,
+                             svfloat32_t &sum) {
+    svbool_t pg = svptrue_b8();
+
+    // Load uint8 vectors
+    svuint8_t v1_ui8 = svld1_u8(pg, pVect1 + offset);
+    svuint8_t v2_ui8 = svld1_u8(pg, pVect2 + offset);
+
+    svfloat32_t ipf32 = svcvt_f32_u32_x(pg, svdot_u32(svdup_u32(0), v1_ui8, v2_ui8));
+
+    sum = svadd_f32_x(pg, sum, ipf32);
+
+    offset += svcntb(); // Move to the next set of uint8 elements
```

Comment:
I wonder if this has a good performance. This function always returns the same value in runtime, but I'm not sure if the compiler can deduce that and only call it once. Have you tried to call it once at the beginning of the function and use its value, instead of calling it here?